ic10emu/src/lib.rs

```diff
@@ -698,7 +719,10 @@ impl VM {
         val: f64,
     ) -> Result<(), ICError> {
         self.batch_device(source, prefab, None)
-            .map(|device| device.borrow_mut().set_slot_field(index, typ, val))
+            .map(|device| {
+                self.set_modified(device.borrow().id);
+                device.borrow_mut().set_slot_field(index, typ, val)
```

Comment:
it we passed in the VM var, yes.
I'm not sure that's a good idea though. I'd rather not taint the direct calls with the vm in case there is a reason to call them during the setup of a device.